### PR TITLE
Move IntelliJ LSP plugin to language support repo.

### PR DESCRIPTION
**Pull request by meistermeier**
This is all WIP right now:
* Figure out how to build this for release
* Sanity check of the configuration/properties
  * More information in the plugin descriptor
* Icon needed :)

### CODE REVIEW

File: editor-plugin/intellij/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        certificateChain.set(System.getenv("CERTIFICATE_CHAIN"))
+        privateKey.set(System.getenv("PRIVATE_KEY"))
+        password.set(System.getenv("PRIVATE_KEY_PASSWORD"))
+    }
+
+    publishPlugin {
+        // alpha, beta or eap
+        channels.set(listOf("beta"))
+        token.set(System.getenv("PUBLISH_TOKEN"))
+    }
```
Comment (ncordon) [inline editor-plugin/intellij/build.gradle.kts:47]: I assume these are needed to release the plugin?

File: editor-plugin/intellij/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Neo4j Cypher LSP support for IntelliJ
+
+## Build and run
+
+* Build the plugin via `./gradlew buildPlugin`.
+* Choose the plugin's zip archive from `build/distributions/cypher-lsp-support-<version>.zip` when installing it manually in IntelliJ
```
Comment (ncordon) [inline editor-plugin/intellij/README.md:6]: Nice 💃. This is only 1.1MB by the way

**Review by ncordon — COMMENTED** [review]
This is awesome 💃 . Only two nitpicks:

1. Is there any way to tell Gradle to copy the `cypher-language-server.js` file from the generated typescript packages instead of duplicating it? 
2. Maybe we can have the neo4j logo as plugin icon?

File: editor-plugin/intellij/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Neo4j Cypher LSP support for IntelliJ
+
+## Build and run
```
Comment (ncordon) [inline editor-plugin/intellij/README.md:3]: Maybe say it needs JDK 17?

File: editor-plugin/intellij/build.gradle.kts (hunk 1 above)
Comment (meistermeier) [reply editor-plugin/intellij/build.gradle.kts:47]: Yes, when/if we plan to release it on the official way. I kept the placeholders to not have to figure those parameters out later.

**Comment by meistermeier** [comment]
> Is there any way to tell Gradle to copy the cypher-language-server.js file from the generated typescript packages instead of duplicating it?

That's where my lack of JS knowledge kicks in: There is no generated version available, right? So the build script would need to create one from the sources and place it there.

> Maybe we can have the neo4j logo as plugin icon?

Yes, I think that Neo4j still represent Cypher the best but couldn't find any svg logo :(

**Comment by OskarDamkjaer** [comment]
I think you can find a neo4j icon here: https://www.neo4j.design/40a8cff71/p/444d31-downloads-overview

**Comment by meistermeier** [comment]
Unfortunately they are all just pngs/jpgs. The only svg is a white full logo. :(

**Review by ncordon — APPROVED** [review]

File: editor-plugin/intellij/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            exec {
+                commandLine("bash", "-c", "rm -rf *.js")
+            }
+        }
+    }
+
+    named("prepareSandbox") {
+        doFirst {
+            exec {
+                commandLine("bash", "-c", "cd ../.. && npm run build && cp packages/language-server/dist/cypher-language-server.js ./editor-plugin/intellij")
```
Comment (ncordon) [inline editor-plugin/intellij/build.gradle.kts:45]: `npm run build` should have it cached if already compiled

File: editor-plugin/intellij/build.gradle.kts [hunk 5]
```diff
@@ -16,38 +16,26 @@ intellij {
...
+    clean {
         doFirst {
             exec {
                 commandLine("bash", "-c", "rm -rf *.js")
             }
         }
     }
 
-    named("prepareSandbox") {
+    prepareSandbox {
```
Comment (meistermeier) [inline editor-plugin/intellij/build.gradle.kts:33]: Looks even more like we know what we are doing now 😄

Comment (ncordon) [reply editor-plugin/intellij/build.gradle.kts:33]: 😂

**Review by ncordon — APPROVED** [review]

**Pull request merged**